src/hotspot/share/opto/c2_globals.hpp

```diff
@@ -82,7 +82,7 @@
           "actual size could be less depending on elements type")           \
           range(0, max_jint)                                                \
                                                                             \
-  product(intx, ArrayCopyPartialInlineSize, -1, DIAGNOSTIC,                 \
+  product(intx, UsePartialInlineSize, -1, DIAGNOSTIC,                       \
```

Comment:
Also, `UsePartialInlineSize` looks like a bool flag, but it's not.
I assume you renamed it because it's not specific to `arraycopy` anymore.
What about `ArrayOperationPartialInlineSize`?